packages/melos/lib/src/common/pub_credential.dart

```diff
@@ -0,0 +1,107 @@
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:cli_util/cli_util.dart';
+import 'package:collection/collection.dart';
+import 'package:meta/meta.dart';
+import 'package:path/path.dart' as path;
+
+import 'io.dart';
+import 'pub_hosted.dart';
+
+const _kPubTokenFileName = 'pub-tokens.json';
+
+@visibleForTesting
+PubCredentialStore pubCredentialStore = PubCredentialStore.fromConfigFile();
+
+PubCredentialStore get pubGlobalCredentialStore => pubCredentialStore;
```

Comment:
Maybe follow the same pattern here as in `http.dart`, so `innerPubCredentialStore` and `pubCredentialStore`?
Maybe `internal` instead of `inner` is more clear?